internal/daemon/daemon.go

```diff
@@ -93,6 +93,9 @@ func New(ctx context.Context, logger logr.Logger, cfg Config) (*Daemon, error) {
 	}
 
 	hostnameService := internal.NewHostnameService(cfg.Host)
+	if cfg.WebhookHost != "" {
+		hostnameService.SetWebhookHostname(cfg.WebhookHost)
+	}
```

Comment:
You might want to take this as an opportunity to refactor the hostname service. Currently, there is hostname "domain logic" here in `daemon.go` that really ought to belong in `hostname.go` instead. The logic I'm talking about is "if this is set, set this, if this is not set, then set that, etc.", which daemon should not care about, but the hostname service _should_ care about.

So have `daemon.go` just set values via the hostname service `Set(...)` methods. The hostname service then determines what to with them, or it's "getter" methods determine which values to return depending on what has been set.

I would do the same with the default hostname, `localhost:8080` - move that into `hostname.go`.

This way, as much as possible on the subject of hostnames is in `hostname.go`.

Reply:
I removed this and moved that logic into `hostname.go`, I think the rest of the refactor could be handled in another PR so this change isn't blocked by that?